app/src/main/java/com/google/homesampleapp/screens/home/HomeFragment.kt

```diff
@@ -383,4 +397,54 @@ class HomeFragment : Fragment() {
     val msgTextView: TextView? = codelabInfoAlertDialog.findViewById(android.R.id.message)
     msgTextView?.movementMethod = LinkMovementMethod.getInstance()
   }
+
+  // ---------------------------------------------------------------------------
+  // Device Attestation Delegate
+
+  private class EmptyAttestationDelegate : DeviceAttestationDelegate {
+    override fun onDeviceAttestationCompleted(
+        devicePtr: Long,
+        attestationInfo: AttestationInfo,
+        errorCode: Int
+    ) {}
+  }
+
+  private fun setDeviceAttestationDelegate() {
+    chipClient.chipDeviceController.setDeviceAttestationDelegate(
+        DEVICE_ATTESTATION_FAILED_TIMEOUT_SECONDS) { devicePtr, attestationInfo, errorCode ->
+          Timber.d(
+              "Device attestation errorCode: $errorCode, " +
+                  "Look at 'src/credentials/attestation_verifier/DeviceAttestationVerifier.h' " +
+                  "AttestationVerificationResult enum to understand the errors")
+
+          if (errorCode == STATUS_PAIRING_SUCCESS) {
+            Timber.d("DeviceAttestationDelegate: Success on device attestation.")
+            lifecycleScope.launch {
+              chipClient.chipDeviceController.continueCommissioning(devicePtr, true)
+            }
+          } else {
+            Timber.d("DeviceAttestationDelegate: Error on device attestation [$errorCode].")
+            // Ideally, we'd want to show a Dialog and ask the user whether the attestation
+            // failure should be ignored or not.
+            // Unfortunately, the GPS commissioning API is in control at this point, and the
+            // Dialog will only show up after GPS gives us back control.
+            // So, we simply ignore the attestation failure for now.
+            // TODO: Add a new setting to control that behavior.
```

Comment:
How about a non-blocking toast once we get control back?

Reply:
IIUC, this toast would simply tell the user that a device attestation failure was ignored, correct? (so that even though user does not have the ability to change the behavior (until we have a setting), at least we make it clear that attestation failure was ignored).

Reply:
Yeah exactly.

Reply:
Fixed in the next commit.

Reply:
Looks good.